src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,205 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.DriveConstants;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+
+public class Drivetrain extends SubsystemBase {
+
+  //Create variables for motor groups and differential drive
+
+  private final CANSparkMax m_leftMotor1;
+  private final CANSparkMax m_leftMotor2;
+  private final CANSparkMax m_rightMotor1;
+  private final CANSparkMax m_rightMotor2;
+
+  private final SpeedControllerGroup m_leftMotors;
+  private final SpeedControllerGroup m_rightMotors;
+
+  private final DifferentialDrive m_Drive;
+
+  //Create variables for encoders
+
+  private final CANEncoder m_leftMotorEncoder1;
+  private final CANEncoder m_leftMotorEncoder2;
+  private final CANEncoder m_rightMotorEncoder1;
+  private final CANEncoder m_rightMotorEncoder2;
+
+  private boolean slow = false;
+
+  private String m_driveType = "Tank"; // adds a String object to store the drive type in
+  SendableChooser<String> m_chooser = new SendableChooser<>();
+
+// Creates a new Drivetrain.
+public Drivetrain() {
+
+    // Set variables equal to their ports and types
+    m_leftMotor1 = new CANSparkMax(DriveConstants.kLeftDriveMotor1Port, MotorType.kBrushless);
+    m_leftMotor2 = new CANSparkMax(DriveConstants.kLeftDriveMotor2Port, MotorType.kBrushless);
+    m_rightMotor1 = new CANSparkMax(DriveConstants.kRightDriveMotor1Port, MotorType.kBrushless);
+    m_rightMotor2 = new CANSparkMax(DriveConstants.kRightDriveMotor2Port, MotorType.kBrushless);
+
```

Comment:
We should be initializing the motors after instantiating their objects.  Set defaults, set inverted or not, set current limit, etc.  Check the Bross-dev branch of the Kairos-rewrite for some ideas.